src/app/(sidebar)/transaction/sign/components/Import.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import { useState } from "react";
+import { Alert, Card, Icon, Text, Button } from "@stellar/design-system";
+import { TransactionBuilder } from "@stellar/stellar-sdk";
+
+import { useStore } from "@/store/useStore";
+
+import { validate } from "@/validate";
+
+import { XdrPicker } from "@/components/FormElements/XdrPicker";
+
+export const Import = () => {
+  const { network, transaction } = useStore();
+  const { updateSignActiveView, updateSignImportXdr, updateSignImportTx } =
+    transaction;
+
+  const [txXdr, setTxXdr] = useState<string>("");
+  const [txErrMsg, setTxErrMsg] = useState<string>("");
+  const [txSuccessMsg, setTxSuccessMsg] = useState<string>("");
+
+  // on textarea onChange for 'import tx xdr'
+  const onChange = (value: string) => {
+    // reset messages onChange
+    setTxErrMsg("");
+    setTxSuccessMsg("");
+    setTxXdr(value);
+
+    if (value.length > 0) {
+      const validatedXDR = validate.xdr(value);
+
+      if (validatedXDR.result === "success") {
+        setTxSuccessMsg(validatedXDR.message);
+      } else {
+        setTxErrMsg(validatedXDR.message);
+      }
+    }
+  };
+
+  const onImport = () => {
+    try {
+      const transaction = TransactionBuilder.fromXDR(txXdr, network.passphrase);
+
+      updateSignImportTx(transaction);
+      updateSignImportXdr(txXdr);
+
+      // change to 'overview' view when successfully imported
+      updateSignActiveView("overview");
+    } catch (e) {
+      setTxErrMsg("Unable to import a transaction envelope");
+      updateSignActiveView("import");
+    }
+  };
+
+  return (
+    <div className="SignTx__Overview">
+      <div className="PageHeader">
+        <Text size="md" as="h1" weight="medium">
+          Sign Transaction
+        </Text>
+      </div>
+      <Card>
+        <div className="SignTx__xdr">
+          <XdrPicker
+            id="sign-transaction-xdr"
+            label={
+              <Text size="xs" as="span" weight="medium">
+                Import a transaction envelope in XDR format
+                <span className="SignTx__icon">
+                  <Icon.AlertCircle />
+                </span>
+              </Text>
+            }
+            value={txXdr || ""}
+            error={txErrMsg}
+            success={
+              <Text size="xs" as="span" addlClassName="success-message">
+                {txSuccessMsg}
+              </Text>
```

Comment:
We might not need this extra styling anymore. The latest SDS version should handle it for ijnputs.